src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,61 @@
 'use strict';
 
+'use strict';
+
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
+
+  isAlive() {
+    return this.health > 0;
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (!this.isAlive()) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
```

Comment:
There is no need to reassign the `name` and `health` properties in the `Herbivore` constructor since they are already assigned in the `Animal` constructor.

Reply:
ok